### PR TITLE
refactor: Rename whitelistFiles to ignorelistFiles and update cspell

### PR DESCRIPTION
**Pull request by gkartalis**
# Description

Hey 👋  I work in Artsy and there was an [RFC](https://github.com/artsy/README/issues/427) for adopting a more inclusive language. We use this plugin and I was thinking maybe we could make a release including the suggested changes below.

This will help us to change the `whitelistFiles` to `ignoreFiles` in some of the `spellcheck.json` files.

Happy to hear your opinion on this 😁.

- [x] renames whitelist to ignorelist everywhere
- [x] make changes backwards compatible
- [x] updates cspell dependency (updated this to latest version since when I forked the project some of the tests were failing due to this error:

```bash RangeError: Maximum call stack size exceeded
        at Function.[Symbol.hasInstance] (<anonymous>)
      at ReadStream (node_modules/configstore/node_modules/graceful-fs/graceful-fs.js:177:14)
      at ReadStream (node_modules/configstore/node_modules/graceful-fs/graceful-fs.js:178:28)
      at ReadStream (node_modules/configstore/node_modules/graceful-fs/graceful-fs.js:178:28)
      at ReadStream (node_modules/configstore/node_modules/graceful-fs/graceful-fs.js:178:28)
      at ReadStream (node_modules/configstore/node_modules/graceful-fs/graceful-fs.js:178:28)
      at ReadStream (node_modules/configstore/node_modules/graceful-fs/graceful-fs.js:178:28)
      at ReadStream (node_modules/configstore/node_modules/graceful-fs/graceful-fs.js:178:28)
      at ReadStream (node_modules/configstore/node_modules/graceful-fs/graceful-fs.js:178:28)
      at ReadStream (node_modules/configstore/node_modules/graceful-fs/graceful-fs.js:178:28)
      at ReadStream (node_modules/configstore/node_modules/graceful-fs/graceful-fs.js:178:28)
      at ReadStream (node_modules/configstore/node_modules/graceful-fs/graceful-fs.js:178:28)
      at ReadStream (node_modules/configstore/node_modules/graceful-fs/graceful-fs.js:178:28)
      at ReadStream (node_modules/configstore/node_modules/graceful-fs/graceful-fs.js:178:28)
      at ReadStream (node_modules/configstore/node_modules/graceful-fs/graceful-fs.js:178:28)
      at ReadStream (node_modules/configstore/node_modules/graceful-fs/graceful-fs.js:178:28)
      at ReadStream (node_modules/configstore/node_modules/graceful-fs/graceful-fs.js:178:28)
      at ReadStream (node_modules/configstore/node_modules/graceful-fs/graceful-fs.js:178:28)
```

### CODE REVIEW

**Comment by orta** [comment]
Yes, but it should be backwards compatible, so under the hood still allow those terms but don't talk about them in the README

**Comment by orta** [comment]
You might also want to think about https://github.com/orta/danger-plugin-spellcheck/issues/30

**Comment by gkartalis** [comment]
Great points 🤔  will address those as well

File: README.md [hunk 1]
```diff
@@ -40,25 +40,25 @@ The JSON should look something like:
 ```json
 {
   "ignore": ["orta", "artsy", "github", "/danger-*."],
-  "whitelistFiles": ["README.md"]
+  "allowlistFiles": ["README.md"]
```
Comment (dblandin) [inline README.md]: **suggestion**: Would an `ignoreFiles` property match better with the existing `ignore` property? In this context, the word "allow" could be interpreted to mean either "allow the tool to check the spelling in these files" or (as is actually the case) "allow any spelling errors in these files".

**Comment by gkartalis** [comment]
> You might also want to think about #30

Don't see any straightforward way to use github api here since if the repo that uses this is private we would need to pass also a github token for each private repo. 🤔  Any suggestions here?

**Comment by gkartalis** [comment]
@orta Maybe we can address #30 on another pr to unblock this?

**Comment by orta** [comment]
Sure, seems reasonable to me

**Comment by orta** [comment]
Shipped as 2.1.0

**Pull request merged**